### PR TITLE
Make mackupApplicationsDir() agnostic to mackup's installation method

### PR DESCRIPTION
**Pull request by ronlut**
Currently the `chezmoi mackup` command assumes `mackup` was installed with brew.
It can be installed with pip and macports as well, so I changed the code to be agnostic to the installation method.

@twpayne Let me know if this PR makes sense from your perspective and if so, I will work on fixing the `mackup_darwin.txtar` test (a hint here on how you would've approached it would be wonderful 😄  )

### CODE REVIEW

**Comment by twpayne** [comment]
Thanks for this PR!

> @twpayne Let me know if this PR makes sense from your perspective and if so, I will work on fixing the `mackup_darwin.txtar` test (a hint here on how you would've approached it would be wonderful 😄 )

Yes, the PR makes sense. I've updated the existing test in #3401 so that the simulated Homebrew installation of mackup is more realistic. Please can you rebase this PR on the latest `master` to get the changes.

It would be awesome if you added similar test for pip and MacPorts installations of mackup. These will likely need to be in separate testscript files, i.e. you'll want to copy the `internal/cmd/testdata/scripts/mackupbrew_darwin.txtar` file and make separate MacPorts and pip equivalents.

**Comment by halostatue** [comment]
`pipx` would be good to simulate as well. It creates a venv per package (`pipx install mackup` would create `$HOME/.local/pipx/venvs/mackup`) and symlinks files to `$HOME/.local/bin` (`$HOME/.local/bin/mackup -> $HOME/.local/pipx/venvs/mackup/bin/mackup`).

For reference, the MacPorts directory would be `/opt/local/bin/mackup -> /opt/local/Library/Frameworks/Python.framework/Versions/3.11/bin/mackup`.

**Comment by ronlut** [comment]
I changed the logic a bit more to support pip installations where the lib folder might contain multiple python versions -> it now looks for the one that has the mackup folder structure in it

File: internal/cmd/testdata/scripts/mackupmacports_darwin copy.txtar
Comment (halostatue) [inline internal/cmd/testdata/scripts/mackupmacports_darwin copy.txtar:1]: This filename should be fixed.

**Comment by halostatue** [comment]
Make sure you run `make format` which should resolve the lint checking failure.

File: internal/cmd/testdata/scripts/mackupmacports_darwin copy.txtar
Comment (ronlut) [reply internal/cmd/testdata/scripts/mackupmacports_darwin copy.txtar:1]: sorry, fixed, don't know how it happened

File: internal/cmd/mackupcmd_darwin.go [hunk 1]
```diff
@@ -122,49 +122,42 @@ func (c *Config) runMackupAddCmd(
 }
 
 func (c *Config) mackupApplicationsDir() (chezmoi.AbsPath, error) {
-	brewPrefixCmd := exec.Command("brew", "--prefix")
-	brewPrefixData, err := brewPrefixCmd.Output()
+	mackupBinaryPath, err := exec.LookPath("mackup")
 	if err != nil {
 		return chezmoi.EmptyAbsPath, err
```
Comment (halostatue) [inline internal/cmd/mackupcmd_darwin.go:127]: I wonder if a better error than `chezmoi: exec: "mackup": executable file not found in $PATH` might be better than this.

`return chezmoi.EmptyAbsPath, fmt.Errorf("Mackup not installed (%s)", err)`?

File: internal/cmd/mackupcmd_darwin.go [hunk 2]
```diff
@@ -122,49 +122,42 @@ func (c *Config) runMackupAddCmd(
...
+		}
+
+		return mackupAppsDir, nil
 	}
 
-	return libDirAbsPath.Join(pythonDirRelPath).
-			JoinString("site-packages", "mackup", "applications"),
-		nil
+	return chezmoi.EmptyAbsPath, fmt.Errorf(
+		"%s: could not find python directory",
```
Comment (halostatue) [inline internal/cmd/mackupcmd_darwin.go:158]: Better error message here, I think. `mackup python directory cannot be found: %s`?

**Review by halostatue — CHANGES_REQUESTED** [review]
Your tests [are failing](https://github.com/twpayne/chezmoi/actions/runs/7159637411/job/19493101490) because of you have not put in any of the simulated `bin/mackup` files (you removed one at line 32 of `mackupbrew_darwin.txtar` (you cannot just put a symlink to a nonexistent file in a `txtar`).

This would need to be applied to all of the `txtar` tests.

Note that the format check test is [continuing to fail](https://github.com/twpayne/chezmoi/actions/runs/7159637411/job/19493099899). `make format` *should* resolve this.

**Comment by ronlut** [comment]
> Your tests [are failing](https://github.com/twpayne/chezmoi/actions/runs/7159637411/job/19493101490) because of you have not put in any of the simulated `bin/mackup` files (you removed one at line 32 of `mackupbrew_darwin.txtar` (you cannot just put a symlink to a nonexistent file in a `txtar`).
> 
> This would need to be applied to all of the `txtar` tests.
> 
> Note that the format check test is [continuing to fail](https://github.com/twpayne/chezmoi/actions/runs/7159637411/job/19493099899). `make format` _should_ resolve this.

It's not, you are referring an old code version before my [latest commit](https://github.com/twpayne/chezmoi/pull/3400/commits/0bf9fefb8256028fb4d1326565e7eb1897e5ba37) yesterday... No actions were run on my last commit for some reason, please rerun.

Anyway, I'll do the changes to error messages you requested and push a new commit later today or tomorrow 👍

**Review by halostatue — APPROVED** [review]

**Review by halostatue — CHANGES_REQUESTED** [review]
The macOS tests failed because of broken symlinks in the txtar files: https://github.com/twpayne/chezmoi/actions/runs/7171249698/job/19525838328

**Comment by ronlut** [comment]
> The macOS tests failed because of broken symlinks in the txtar files: https://github.com/twpayne/chezmoi/actions/runs/7171249698/job/19525838328

Okay after some more digging I understood why it passed locally and not in the ci: the `exec.LookPath` finds the mackup installed in my system, instead of the txtar one... The symlink wasn't the problem, it was ok, the problem was that it didn't pass the `findExecutable` in `LookPath` and moved to the next directory in `PATH` and found the system one. 
Anyways :)
I think now it should be ok. Maybe worth thinking of a way to execute the tests in a more isolated way locally.

**Comment by halostatue** [comment]
Thank you for your contribution, @ronlut!

**Comment by halostatue** [comment]
Sorry. I forgot to have the history cleaned up before merging. :D

**Pull request merged**